core/domain/user_services.py

```diff
@@ -551,3 +560,28 @@ def get_user_impact_score(user_id):
         return model.impact_score
     else:
         return 0
+
+def fetch_gravatar(email):
+    """Returns the gravatar corresponding to the user's email,
+
+    or an default image if gravtar is not receieved
+    """
+    default_identicon_data_url = feconf.DEFAULT_IDENTICON_DATA_URL
+    base_url = 'http://www.gravatar.com/avatar/'
+    gravatar_url = base_url + hashlib.md5(email).hexdigest() + '?'
+    params = {'d':'identicon', 's':feconf.GRAVATAR_SIZE_PX}
+    gravatar_url += "&".join("%s=%s" % (k, v) for k, v in params.items())
+    try:
+        result = urlfetch.fetch(
+            gravatar_url,
+            headers={'Content-Type': 'image/png'},
+            follow_redirects=False)
+        if result.status_code == 200:
+            encoded_body = base64.b64encode(result.content)
```

Comment:
See ImageUploadHandler in core/controllers/editor.py -- maybe some of that logic can be abstracted out (into, say, utils.py).